### ISSUE TITLE
[v0.4.5.3] Latest release does not correctly install via pip (master seems fixed!)

### ISSUE THREAD

**Issue by e-d-n-a** [bubblesub/pyffms2#24]
The current version of the [latest release](https://github.com/bubblesub/pyffms2/releases/tag/0.4.5.3) and [PyPI-package ](https://pypi.org/project/ffms2/)is v0.4.5.3!

**Installation:**
```
>pip install ffms2
Collecting ffms2
  Downloading ffms2-0.4.5.3.tar.gz (692 kB)
     |████████████████████████████████| 692 kB 6.4 MB/s
Requirement already satisfied: numpy in c:\winpython\wpy-x64-3870\python-3.8.7.amd64\lib\site-packages (from
 ffms2) (1.19.4+mkl)
Collecting pypiwin32
  Downloading pypiwin32-223-py3-none-any.whl (1.7 kB)
Requirement already satisfied: pywin32>=223 in c:\winpython\wpy-x64-3870\python-3.8.7.amd64\lib\site-package
s (from pypiwin32->ffms2) (300)
Building wheels for collected packages: ffms2
  Building wheel for ffms2 (setup.py) ... done
  Created wheel for ffms2: filename=ffms2-0.4.5.3-py3-none-any.whl size=707644 sha256=85af4f3aae7e1f05ea85f9
d4bd8d673a6b5440b960e250ffdfcf84f3022e8d59
  Stored in directory: c:\users\xxx\appdata\local\pip\cache\wheels\db\4d\8d\1d06c2bb2ab7970b9b651ee06df1cd8c4
e3cbae800b73467e6
Successfully built ffms2
Installing collected packages: pypiwin32, ffms2
Successfully installed ffms2-0.4.5.3 pypiwin32-223
```

**Errors while running test.py:**
[see in issues below!]

**There are 3 major issues with this release, that I opened as separate issues, so that they can be addressed individually:**

1. no wheels (including ffms2-lib) for Python other than v3.7-x64, see #25
2. dependency is "pypiwin32" instead of "pywin32", see #26
3. import-issue in ffms_init()/ffms_uninit() of \_\_init\_\_.py, see #27

**It was not to hard to get it fixed myself, but the release should be corrected so it can be installed correctly!**

I installed and tested on Windows 7 SP1-x64 within a fresh WinPython-Installation ("python-3.8.7.amd64" with default set of packages preinstalled including "pywin32")

**Comment by rr-** [comment]
@e-d-n-a can you confirm if the newest release meets your requirements?

**Comment by e-d-n-a** [comment]
### I tested with two different environments ("WPy" = "WinPython"):

1. **WPy-x64-3760ps2** [PySide2-Version]
Has many packages preinstalled (including "numpy" and "pywin32")

2. **WPy-x64-3870Z** [Zero-Version]
Has only installation packages pre-installed.
```
Package      Version
------------ -------------
msvc-runtime 14.28.29325
pip          20.3.3
setuptools   51.1.0
sqlite-bro   0.9.1
wheel        0.36.2
winpython    3.0.202011219
```

In both cases "**[ffms2-0.4.5.5-cp37.cp38.cp39-none-win_amd64.whl](https://files.pythonhosted.org/packages/e4/d5/bf83d494a1ade8315908e11324ad9c35e7f88d744b51e8a95984a6dbe788/ffms2-0.4.5.5-cp37.cp38.cp39-none-win_amd64.whl)**" was used for the installation and proceeded without errors.
The ffms2-binaries get installed into the "Scripts"-folder of the Python-installation as intended (I guess).

So, in case 1 the dependencies are already available and only the "ffms2" package gets installed and it works directly, when tested via a Test-Script of mine. So no "pypiwin32" got installed this time (obviously).

### Yet, in case 2 a problem can still be seen:
The "ffms2"-package gets installed fine and also directly declared dependencies are installed ("numpy"),
while **"pywin32" does NOT get installed**, because the dependency is only injected via the "[setup.py](https://github.com/bubblesub/pyffms2/blob/master/setup.py)"-script, that exists in the source package, when installing manually on Windows.

As the Wheel seems to be Windows-specific, you should add the "pywin32" dependency in it!
I have no experience with building Python packages/Wheels, but I can [see here](https://github.com/bubblesub/pyffms2/blob/73ff03a468cddf8c8c1c8cb2781d5c215a169cb7/.github/workflows/deploy-binaries.yml#L273), that you use the "[setup.py](https://github.com/bubblesub/pyffms2/blob/master/setup.py)"-script to build the Wheel.
This script doesn't seem to do the job equally well for a manual installation and for building the Wheel.
**As you're building [on Linux "pywin32" won't get added as a dependency](https://github.com/bubblesub/pyffms2/blob/73ff03a468cddf8c8c1c8cb2781d5c215a169cb7/setup.py#L8) for the Wheel.**

You may want to consider switching to a "setup.cfg"-based approach, as it seems the modern way to configure the installation.

Checkout the [setuptools-documentation for platform-specific dependencies](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies).

Otherwise, a simple fix for "[setup.py](https://github.com/bubblesub/pyffms2/blob/master/setup.py)" should be:

```
if set(["bdist_wheel", "--plat-name", "win_amd64"]) <= set(sys.argv) or (
    os.name == "nt" and all([os.path.exists(f) for f in libs])
):
    data_libs = [("Scripts", libs)]
    install_packages.append("pywin32")
```

**Comment by e-d-n-a** [comment]
In addition of being in "[Python-Installation]\Lib\site-packages\ffms2-0.4.5.5.dist-info" the files "COPYING" and "COPYING.LESSER" end up in "[Python-Installation]\Lib\site-packages" as well btw.

I think this is not correct!

Why are they added to "package_data" [here](https://github.com/bubblesub/pyffms2/blob/73ff03a468cddf8c8c1c8cb2781d5c215a169cb7/setup.py#L40)!?

Check out these links:
https://stackoverflow.com/questions/9977889/how-to-include-license-file-in-setup-py-script
https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html#metadata
https://github.com/pypa/wheel/issues/ [#138] "Include more than one license file or other arbitrary files in metadata"

**Comment by Luni-4** [comment]
Thanks for all your tests @e-d-n-a!  :)

To fix the first issue, we can't use this solution:
> ```
>     install_requires=[
>         "numpy",
>         "pywin32;platform_system=='Windows'"
>         ]
> ```

because of [this](https://github.com/bubblesub/pyffms2/commit/53be0e2bb7bab8afc54cba34b5dae690cb2a0352) problem. We would break mingw compilation. So it would be better to update `pypiwin32` to `pywin32` everywhere. I don't know if there is a better way to detect when the package is being built with `mingw` though

About this one: 

> In addition of being in "[Python-Installation]\Lib\site-packages\ffms2-0.4.5.5.dist-info" the files "COPYING" and "COPYING.LESSER" end up in "[Python-Installation]\Lib\site-packages" as well btw.
> 
> I think this is not correct!
> 
> Why are they added to "package_data" [here](https://github.com/bubblesub/pyffms2/blob/73ff03a468cddf8c8c1c8cb2781d5c215a169cb7/setup.py#L40)!?
> 
> Check out these links:
> https://stackoverflow.com/questions/9977889/how-to-include-license-file-in-setup-py-script
> https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html#metadata
> https://github.com/pypa/wheel/issues/ [#138] "Include more than one license file or other arbitrary files in metadata"
> https://pythonwheels.com/ [some interesting info about Wheels]

What do you think @rr-? Should we remove the licenses from the package? It's the same for me, so feel free to adopt the method you consider more appropriate.

**Comment by Luni-4** [comment]
For what concerns the `setup.cfg`, I agree, we should switch to that. The code would be much clear, but I've never have done that and can't work on it in the short period.

**Comment by rr-** [comment]
I was contacted before by Linux distributions maintainers that the packages need to contain their licenses.

**Comment by Luni-4** [comment]
I see, so only the first issue remains. To fix it, we could mix the two checks of line 7 and 14 together.

**Comment by e-d-n-a** [comment]
The problem is not about removing the licenses, but to declare them correctly so they only end up in "ffms2-0.4.5.5.dist-info".
But maybe only the declarative config supports the "license_files" metadata key, that would be the correct (generic) solution.

Yet, it seems ["bdist_wheel" automatically includes license files with the pattern "COPYING*"](https://github.com/pypa/wheel/blob/41b375ef33e3368f6f58c13f9649491d01c2ba5e/src/wheel/bdist_wheel.py#L422).
So you at least should not include them again via "package_data", when building the wheel.

**Comment by Luni-4** [comment]
Fixed in 4f68476
